---
title: "sctld_RNAseq_DESeq"
author: "Mike Connelly"
date: "11/4/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r packages}
# install.packages("tidyverse", "DESeq2", "pheatmap")
library("tidyverse")
library("here")
library("tximport")
library("DESeq2")
# extended visualization functions
source(here("R", "custom_viz_functions.R"))
```

```{r import_data}
samples <- read_csv(here("data", "metadata.csv"))
# 
samples <- read_csv(here("data", "metadata_pcliv.csv"), 
                    col_types = cols(
  K_ID = col_character(),
  Tube_Label = col_logical(),
  Project = col_character(),
  Coral_species = col_character(),
  Date_Collected = col_character(),
  Coral_Genotype = col_character(),
  pct_progress = col_double(),
  Transmission_species = col_logical(),
  Treatment = col_character(),
  X10 = col_logical()
))
```

```{r tximport}
quant_folders <- str_c(samples$K_ID, "_quant")
quant_files <- file.path(here("outputs", "salmon_quants", quant_folders), pattern = "quant.sf")
quant_files
txi <- tximport(quant_files, type="salmon",
                txIn = T, txOut = T,
                countsFromAbundance = "no")
?tximport
```
```{r transcript_annotation}
# gene_annotation <- read.delim(file = "./data/psst_annotations.tsv", header = T) %>% arrange(ID)
# rownames(gene_annotation) <- gene_annotation$ID
# # Check gene feature annotation and countdata rowname order coherence
# all(rownames(countdata.sorted) == gene_annotation$ID)
# all(rownames(countdata.sorted) ==  rownames(gene_annotation))
# # Obtain KOG annotations for P. damicornis genome
# gene2kog <- gene_annotation %>%
#   dplyr::select(ID, KOG_Class) %>% 
#   filter(KOG_Class != "")
```

```{r create_DESeq_dataset}
ddsTxi <- DESeqDataSetFromTximport(txi,
                                   colData = samples,
                                   design = ~ Treatment)
```
```{r inspect_DESeq_dataset}
# Check annotation and dds object rowname order coherence
# all(rownames(dds) == rownames(gene_annotation))
# # Add gene feature annotation to DESeqDataSets
# mcols(dds) <- cbind(mcols(dds), gene_annotation)
# Subset DESeqDataSet
# Remove genes with counts less than 10 in 90% of samples
keep_tencounts <-  rowSums(counts(ddsTxi)) >= 10
keep_ninety <- rowSums(counts(ddsTxi) >= 10) > (6*0.9)
ddsTxi <- ddsTxi[keep_ninety, ]
```
```{r vst}
# Normalize expression data for visualization purposes using VST tranformation
vsd <- vst(ddsTxi, blind = TRUE) # use blind = TRUE to not account for experimental design
```
```{r}
plotPCA(vsd, intgroup = "Treatment")
```

```{r}
ggPCA(vsd, samples, pclab = c(1,3))
```
```{r mds, eval = TRUE}
# Calculate distances among samples
sampleDists <- dist(t(assay(vsd)), method = "manhattan")
sampleDistMatrix <- as.matrix(sampleDists)

# Calculate MDS
mds <- as.data.frame(colData(vsd)) %>% 
  cbind(cmdscale(sampleDistMatrix))

# Calculate MDS and use eigenvectors to determine proport
mds_eig <- cmdscale(sampleDistMatrix, eig = TRUE)
mds_eigenvectors <- data.frame(mds_eig$eig) %>% 
  mutate(prop_var = mds_eig.eig / sum(mds_eig.eig))
```
```{r treatment pcoa}
# Calculate Treatment centroids for plotting
mds_trmt <- mds %>%
  group_by(Treatment) %>%
  dplyr::summarise(c1 = mean(`1`), c2 = mean(`2`)) %>%    
  full_join(mds)
#set factor orders 
# mds_trmt$Colony <- factor(mds_trmt$Colony, levels = c("HW1", "HW2", "WT1", "WT2"), ordered = TRUE)
mds_trmt$Treatment <- factor(mds_trmt$Treatment, levels = c("Control", "Experimental"), ordered = TRUE)

# Plot with spiders
pcoa2 <- ggplot(mds_trmt, aes(fill = Treatment)) +
  # treatment ellipses
  # stat_ellipse(aes(x = `1`, y = `2`, color = Treatment, fill = Treatment), geom = "polygon", type = "norm", alpha = 0.0) + 
  # sample-centroid spiders paths
  # geom_segment(mapping = aes(x = `1`, y = `2`, xend = c1, yend = c2),
               # lwd = 0.25, col = "dark grey") +
  # treatment centroid points
  geom_point(size = 3, aes(x = c1, y = c2, color = Treatment), fill = "black", shape = 21, stroke = 2, show.legend = FALSE) +
  # sample points
  geom_point(size = 3, aes(x = `1`, y = `2`, color = Treatment), stroke = 0.5, show.legend = FALSE)
  # scale_color_manual(values = condcolors_AxH) +
  # scale_fill_manual(values = condcolors_AxH) +
  # scale_shape_manual(values = colshapes) +
  # labs(x = xlab, y = ylab)
pcoa2
```

```{r DESeq_analysis}
design(ddsTxi) <- formula(~ Treatment)
#Set control treatment as reference factor level
# ddsTxi$Treatment <- factor(ddsTxi$Treatment, levels = c("Control", "Experiment"))
#Perform DESeq2 analysis
dsr1 <- DESeq(ddsTxi)
resultsNames(dsr1)
```

```{r DESeq_contrasts}
res <- results(dsr1, name="Treatment_Experimental_vs_Control")
summary(res)
```


## DESeq results visualizations

```{r DESeq_built-in_functions}

```
```{r pca}

```

```{r volcano_plot}

```

```{r heatmaps}
library("pheatmap")
library("ComplexHeatmap")

```

